src/backend/utils/misc/guc.c

```diff
@@ -2883,7 +2883,7 @@ static struct config_string ConfigureNamesString[] =
 			GUC_LIST_INPUT | GUC_LIST_QUOTE
 		},
 		&temp_spill_files_tablespaces,
-		"",
+		TEMP_TABLESPACES_FALLBACK_VALUE,
```

Comment:
Why not NULL?

Reply:
We should be able to `SET` this value on segments. We can't set NULLs like that

Reply:
> We should be able to `SET` this value on segments

```sql
select current_setting('temp_tablespaces') = '';
 ?column? 
----------
 t
(1 row)

set temp_tablespaces = '';
SET
select current_setting('temp_tablespaces') = '';
 ?column? 
----------
 f
(1 row)

select current_setting('temp_tablespaces') = '""';
 ?column? 
----------
 t
(1 row)

select  set_config('temp_tablespaces', '', false);
 set_config 
------------
 
(1 row)

select current_setting('temp_tablespaces') = '';
 ?column? 
----------
 t
(1 row)
```

Reply:
> We should be able to `SET` this value on segments.

GUC `timezone_abbreviations` has NULL as boot value:

```sql
show timezone_abbreviations;
 timezone_abbreviations 
------------------------
 Default
(1 row)

set timezone_abbreviations = 'Australia';
SET
show timezone_abbreviations;
 timezone_abbreviations 
------------------------
 Australia
(1 row)

set timezone_abbreviations = DEFAULT;
SET
show timezone_abbreviations;
 timezone_abbreviations 
------------------------
 Default
(1 row)
```

Reply:
Are you suggesting using `set_config` for GUC sync instead of `SET`? If it doesn't quote data it can indeed be a general solution.

> GUC timezone_abbreviations has NULL as boot value

It gets initialized to Default [later](https://github.com/arenadata/gpdb/blob/ca1ebaad090bb7e6ea4769553dcba35463f4a07c/src/backend/utils/misc/guc.c#L9931).